### DISCUSSION TITLE
Dropping support for EOL Ruby

### DISCUSSION THREAD

**Discussion by bobbrodie** [sumoheavy/jira-ruby#440] [Announcements]
# Overview

The library was initially built against older versions of Ruby, but now that we're modernizing, I propose dropping support for any EOL versions.

<img width="100%" alt="Screenshot 2024-04-01 at 7 58 42 PM" src="https://github.com/sumoheavy/jira-ruby/assets/123703/2b3aa77b-1341-4b67-918f-f4c3e2735cb0">

Reference: ([link](https://www.ruby-lang.org/en/downloads/branches/))

This would mean adding support for 3.2 and 3.3, and dropping 2.7 and below (should include 3.0 as well, but it did not EOL yesterday).

# Required Updates

Most importantly, we'll need to consider a new major version (`v3.0.0`) due to the backward incompatibilities. We'll also need to:
- Check dependencies (perhaps drop `guard` in favor of something else, like `pre-commit`)
- Resolve rake issues
- Update syntax where mandatory (and other optimizations, with major changes saved for later)

**Comment by marlinpierce** [comment]
I did like the charm of the version being the year.  Version 0.2.2 for 2022, 0.2.3 for 2023.

It also sets expectations.

**Comment by marlinpierce** [comment]
Is there indeed backward incompatibility?  I have been working on a number of branches for the PRs with ruby 2.7 and calling the jira-ruby gem the same way we coded for the prior version.

**Comment by bobbrodie** [comment]
@marlinpierce the incompatibility comes into play with the update to the 1.0 release of the OAuth gem. Here's the build: https://github.com/sumoheavy/jira-ruby/actions/runs/8514256166. The author is dropping non-commercial support for 0.6.x this month https://github.com/oauth-xx/oauth-ruby/releases.

That said, I definitely want to be supportive here! I could branch off the most recent tag, merge some of the updates into it, and release that as 2.4.0 in the interim. That way we can keep it semver -- there was no intent to matching the release date by year.

**Comment by marlinpierce** [comment]
Oh, I wasn't expressing a need for backward compatibility.  I was just asking for an explanation of what is incompatible.

**Comment by bobbrodie** [comment]
Unpinning and moving to discussions.
